substrate/frame/revive/src/precompiles/builtin/create2.rs

```diff
@@ -0,0 +1,89 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use crate::{
+	address::AddressMapper,
+	precompiles::{BuiltinAddressMatcher, BuiltinPrecompile, Error, ExtWithInfo},
+	CodeInfoOf, Config, H256,
+};
+use alloc::{vec, vec::Vec};
+use alloy_core::sol;
+use core::{marker::PhantomData, num::NonZero};
+use sp_arithmetic::traits::SaturatedConversion;
+use sp_core::U256;
+use sp_runtime::DispatchError;
+
+sol! {
+	interface ICreate2 {
+		function create2(bytes32 salt, bytes32 code_hash) external payable returns (address);
+	}
+}
```

Comment:
I think I commented before we can't use the sol! trait here as it means that the call will be prefixed with the 4 bytes hash of the function, which the original yul code linked to the issue and used by foundry does not use

Reply:
How is this possible? A normal create2 factory which is deployed as a contract will do the same? It is written in Solidity and hence expects the 4 byte selector.

Reply:
it's written in yul here
https://github.com/Arachnid/deterministic-deployment-proxy/blob/master/source/deterministic-deployment-proxy.yul